### ISSUE TITLE
[P2][algorithm/classification] RsClassificationPipeline and RsCrossValidation assign unmapped cluster IDs to class output when KMeans clusters exceed ground-truth classes

### ISSUE THREAD

**Issue by WindWang2** [WindWang2/exp-rs#422]
## Description

In `RsClassificationPipeline::run` and `RsCrossValidation::kFold`, supervised KMeans classification matches KMeans clusters to ground-truth training classes via the Hungarian assignment algorithm. When the number of clusters exceeds the number of ground truth classes ( > N$, e.g., =5$ clusters but only =2$ training classes like `{10, 20}`), the Hungarian algorithm matches 2 clusters and leaves the remaining 3 clusters unmapped.

During tile inference and cross-validation, unmapped cluster IDs fall back to `kmeansRemap.value(v, v)` instead of being assigned to `unclassifiedValue` (0). Consequently, pixels belonging to unmapped clusters are written into the output classification raster as their raw integer cluster index (e.g. `3, 4, 5`), creating fabricated, unknown class IDs in the output GeoTIFF palette and corrupting accuracy assessment metrics.

---

## Root Cause Analysis

In `src/analysis/classification/rs_classification_pipeline.cpp`:
1. Lines 466–473 populate `kmeansRemap` only for matched clusters:
```cpp
for ( int i = 0; i < N && i < assign.size(); ++i )
{
  const int clusterIdx = assign[i];
  if ( clusterIdx >= 0 && clusterIdx < M )
    kmeansRemap[cList[clusterIdx]] = tList[i];
}
```
2. When  > N$, unassigned clusters in `cList` are never inserted into `kmeansRemap`.
3. At line 950:
```cpp
v = kmeansRemap.value( v, v );
```
If a pixel is assigned to cluster 3, `kmeansRemap.value(3, 3)` evaluates to `3`. Class ID 3 was never in the user's training set `{10, 20}`.
4. In `src/analysis/classification/rs_cross_validation.cpp` (line 204):
```cpp
yp.append( kmeansRemap.value( pred, pred ) );
```
Test samples falling in unassigned clusters are evaluated as class ID `pred`, inserting phantom columns into the confusion matrix.

---

## Proposed Fix

1. In `RsClassificationPipeline::run`, populate unmatched cluster indices in `kmeansRemap` with `config.ignoreOptions.unclassifiedValue` (0), or replace line 950 with:
```cpp
v = kmeansRemap.value( v, config.ignoreOptions.unclassifiedValue );
```
2. In `RsCrossValidation::kFold` (line 204), replace `kmeansRemap.value( pred, pred )` with:
```cpp
yp.append( kmeansRemap.value( pred, 0 ) );
```

**Comment by WindWang2** [comment]
Resolved and verified in master branch (commit f3661f7f64 and prerequisite commits). All tests passing.
